kafka-rest/src/main/java/io/confluent/kafkarest/response/StreamingResponse.java

```diff
@@ -59,6 +62,7 @@
 public abstract class StreamingResponse<T> {
 
   private static final Logger log = LoggerFactory.getLogger(StreamingResponse.class);
+  @VisibleForTesting static ChunkedOutputFactory factory;
```

Comment:
We can:
1. Create a (default-visibility) `StreamingResponseFactory` that takes a `ChunkedOutputFactory` in the constructor, and has the `from(MappingIterator)` method.
2. Add a `ChunkedOutputFactory` to the StreamingResponse constructors. Pass it down from the `StreamingResponseFactory`.
3. Have our static `from(MappingIterator)` use a static final `StreamingResponseFactory`.
4. Enjoy our life free of mutable static variables. :)